cmd/thanos/bucket.go

```diff
@@ -213,4 +217,118 @@ func registerBucket(m map[string]setupFunc, app *kingpin.Application, name strin
 			return printBlock(id)
 		})
 	}
+
+	inspect := cmd.Command("inspect", "inspect all blocks in the bucket")
+	replica := inspect.Flag("replica ", "Filter by replica.").
+		Default("").String()
+	sortBy := inspect.Flag("sort-by", "Sort by columns.").
+		Default("FROM,UNTIL").String()
+
+	m[name+" inspect"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ bool) error {
+		bucketConfig, err := objStoreConfig.Content()
+		if err != nil {
+			return err
+		}
+
+		bkt, err := client.NewBucket(logger, bucketConfig, reg, name)
+		if err != nil {
+			return err
+		}
+
+		// Dummy actor to immediately kill the group after the run function returns.
+		g.Add(func() error { return nil }, func(error) {})
+
+		defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+
+		// Getting Metas
+		var blockMetas []*block.Meta
+		err = bkt.Iter(ctx, "", func(name string) error {
+			id, ok := block.IsBlockDir(name)
+			if !ok {
+				return nil
+			}
+
+			m, err := block.DownloadMeta(ctx, logger, bkt, id)
+			if err != nil {
+				return err
+			}
+
+			blockMetas = append(blockMetas, &m)
+
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
+
+		return printTable(blockMetas, replica, sortBy)
+	}
+}
+
+func printTable(blockMetas []*block.Meta, replica *string, sortBy *string) error {
+	header := []string{"ULID", "FROM", "UNTIL", "RANGE", "UNTIL-COMP", "~SIZE", "#SERIES", "#SAMPLES", "#CHUNKS", "COMP-LEVEL", "COMP-FAILED", "REPLICA", "RESOLUTION", "SOURCE"}
+
+	var lines [][]string
+	p := message.NewPrinter(language.English)
+
+	for _, blockMeta := range blockMetas {
+		if *replica != "" && *replica != blockMeta.Thanos.Labels["replica"] {
```

Comment:
This is too tied to what user specified as replica label. Probably should be configured

Reply:
I added a labels column which can be filtered via --selector or -l (similar to kubectl get po -l key=value)